diffrax/integrate.py

```diff
@@ -768,7 +768,7 @@ def _promote(yi):
     save_index = 0
     made_jump = False if made_jump is None else made_jump
     ts = jnp.full(out_size, jnp.inf)
-    ys = jax.tree_map(lambda y: jnp.full((out_size,) + jnp.shape(y), jnp.inf), y0)
+    ys = jax.tree_map(lambda y: jnp.full((out_size,) + jnp.shape(y), jnp.inf + 0.0j), y0)
```

Comment:
This will always be complex. Most of the time we will still want something real, so there should be a check here, e.g:
```python
if jnp.iscomplexobj(y0):
    fill_val = jnp.inf + 0.0j
else:
    fill_val = jnp.inf
```